Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+/*
+===========================================================
+  SIMPLE CONTACT BOOK APPLICATION
+===========================================================
+In this project, you'll create a Contact Book to store and
+manage basic info: name, phone, and email.
+
+You'll practice:
+1. Arrays and objects
+2. Loops (for, for-of, findIndex, etc.)
+3. Conditionals (if-else)
+4. Basic CRUD (Create, Read, Update, Delete) functionality
+
+Below is a step-by-step guide, with sample code and
+comments explaining what each section does. Run this file
+in Node.js or in your browser's console to see the output.
+*/
+
+/*
+-----------------------------------------------------------
+  STEP 1: Setup and Initial Data
+-----------------------------------------------------------
+1. Create an array named 'contacts' with a few initial
+   sample contacts.
+2. Each contact is an object with 'name', 'phone', and
+   'email' properties.
+*/
+console.log('STEP-1: Setup and Initial Data');
+
+let contacts = [
+  {
+    name: 'Alice',
+    phone: '123-456-7890',
+    email: 'alice@gmail.com'
+  },
+  {
+    name: 'Anton',
+    phone: '123-456-7899',
+    email: 'anton@gmail.com'
+  },
+  {
+    name: 'Bob',
+    phone: '222-222-2222',
+    email: 'bob@gmail.com'
+  }
+];
+console.log(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 2: Display All Contacts
+-----------------------------------------------------------
+Function: displayAllContacts()
+- Loops over the 'contacts' array.
+- Logs a descriptive string for each contact.
+
+Example output:
+  Name: Alice, Phone: 123-456-7890, Email: alice@example.com
+*/
+console.log('STEP 2: Display All Contacts');
+console.log('Method-1');
+
+function displayAllContacts(contactList){
+
+  for(let contact of contactList){
+    console.log(`Name: ${contact.name}, Phone: ${contact.phone}, Email: ${contact.email}`);
+  }
+}
+displayAllContacts(contacts);
+
+console.log('Method-2');
+
+function displayAllContactsUppercase(contactList, uppercaseMode){
+  for (const contact of contactList){
+    if(uppercaseMode){
+      console.log(`NAME: ${contact.name}, PHONE: ${contact.phone}, EMAIL: ${contact.email}`,);
+      continue;
+    }else{
+      console.log(`name: ${contact.name}, phone: ${contact.phone}, email: ${contact.email}`,);
+    }
+  }
+}
+displayAllContactsUppercase(contacts, true);
+displayAllContactsUppercase(contacts, false);
+
+/*
+-----------------------------------------------------------
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+console.log('STEP 3: Add a New Contact');
+
+
+let contacts3 = [
```

Comment:
You can have multiple contact lists to test your application with. In this case, move all variables holding arrays of contacts to the top of the file so your code is better organized.